src/Datadog.Trace.ClrProfiler.Managed.Loader/StartupLogger.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.IO;
+
+namespace Datadog.Trace.ClrProfiler.Managed.Loader
+{
+    internal static class StartupLogger
+    {
+        private const string NixDefaultDirectory = "/var/log/datadog/dotnet";
+
+        private static readonly string LogDirectory = GetLogDirectory();
+        private static readonly string StartupLogFilePath = LogDirectory != null ? Path.Combine(LogDirectory, "dotnet-tracer-loader.log") : null;
+
+        public static void Log(string message, params object[] args)
+        {
+            try
+            {
+                if (StartupLogFilePath != null)
+                {
+                    try
+                    {
+                        using (var fileSink = new FileSink(StartupLogFilePath))
+                        {
+                            fileSink.Info($"[{DateTime.UtcNow}] {message}{Environment.NewLine}", args);
+                        }
+
+                        return;
+                    }
+                    catch
+                    {
+                        // ignore
+                    }
+                }
+
+                Console.WriteLine(message, args);
+            }
+            catch
+            {
+                // ignore
+            }
+        }
+
+        public static void Log(Exception ex, string message, params object[] args)
+        {
+            message = $"{message}{Environment.NewLine}{ex}";
+            Log(message, args);
+        }
+
+        private static string GetLogDirectory()
+        {
+            string logDirectory = null;
+
+            try
+            {
+                var nativeLogFile = Environment.GetEnvironmentVariable("DD_TRACE_LOG_PATH");
+
+                if (!string.IsNullOrEmpty(nativeLogFile))
+                {
+                    logDirectory = Path.GetDirectoryName(nativeLogFile);
+                }
+
+                if (logDirectory == null)
+                {
+                    var windowsDefaultDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), @"Datadog .NET Tracer", "logs");
+                    if (Directory.Exists(windowsDefaultDirectory))
```

Comment:
Can we be more rigorous in this check? If a user deletes the `logs` directory, we will go down the Linux path.

Reply:
Also, users can change the logs directory with an environment variable.

Reply:
I did have the "IsWindows" check, but it required adding a dependency.
I figured that a directory not existing for each check was a "good enough" check.
What would you feel comfortable with?

Reply:
Added `public static bool IsWindows`
Thoughts?

Reply:
That's better, although I don't think we need to property for a simple enum comparison. Resolving this comment since you already made the change. Thanks!